### PR TITLE
Issue #80: Enhancing `add` command

### LINKED ISSUE #80: Further enhancement: `add` command
- [x] To check for duplicate entries based on `studentId`
- [x] Validate course input e.g. `Business Analytics`, `Computer Engineering`, `Computer Science`, `Information Security`, `Information Systems`
- [x] Reinforce all `Candidate` fields and ensure it's bug-free
- [x] Preset default values for statuses i.e. `Application Status`, `Interview Status` - Completed in PR #87.

### PR DESCRIPTION
**Pull request by domlimm**
This PR consists of the following changes:
- Refactored check for duplicate records to use only `studentId`
- Added deeper validation for `course` input
- Refactored `ApplicationStatus` and `InterviewStatus` attribute and constructor
- Fixed test cases
- Added minor refactors to feedback displayed to user

Closes #80.

### CODE REVIEW

File: src/main/java/seedu/address/model/candidate/InterviewStatus.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
     public static final String INTERVIEWING_STATUS = "Interviewing";
     public static final String NOT_INTERVIEWED_STATUS = "Not Interviewed";
 
-    private String interviewStatus;
+    public final String interviewStatus;
```
Comment (tiewweijian) [inline src/main/java/seedu/address/model/candidate/InterviewStatus.java:11]: Perhaps this one should remain as private - just to have a better level of abstraction. But smol change, I can do it in the next PR ! Thanks dom!

File: src/main/java/seedu/address/model/candidate/ApplicationStatus.java [hunk 2]
```diff
@@ -5,10 +5,10 @@
     public static final String MESSAGE_CONSTRAINTS =
             "Application Status should be Pending, Accepted or Rejected";
     public static final String PENDING_STATUS = "Pending";
-    public static final String ACCPETED_STATUS = "Accepted";
+    public static final String ACCEPTED_STATUS = "Accepted";
```
Comment (tiewweijian) [inline src/main/java/seedu/address/model/candidate/ApplicationStatus.java:8]: thanks for catching my english mistake !! appreciate it!

**Review by tiewweijian — APPROVED** [review]
Thanks for making the minor tweaks to the codebase ! I think codecov/patch is failing because the changes are too minor, but should still be able to merge in ! LGTM ! Let's wait for @LeeEnEn's opinion for the time being.

File: src/main/java/seedu/address/model/candidate/ApplicationStatus.java (hunk 2 above)
Comment (domlimm) [reply src/main/java/seedu/address/model/candidate/ApplicationStatus.java:8]: just a minor typo 😛

File: src/main/java/seedu/address/model/candidate/InterviewStatus.java (hunk 1 above)
Comment (domlimm) [reply src/main/java/seedu/address/model/candidate/InterviewStatus.java:11]: yes please!

File: src/main/java/seedu/address/model/candidate/Course.java [hunk 3]
```diff
@@ -29,7 +32,17 @@ public Course(String course) {
      * Returns true if a given string is a valid course.
      */
     public static boolean isValidCourse(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (test.length() > MAX_LENGTH) {
+            return false;
+        }
+
+        for (int index = 0; index < COURSES.length; ++index) {
+            if (test.equals(COURSES[index]) && test.length() == LENGTHS[index]) {
```
Comment (LeeEnEn) [inline src/main/java/seedu/address/model/candidate/Course.java:40]: Perhaps the user input `test` and the courses to match `COURSES[index]` should be lower case? Makes inputting to the `add command` easier as one would not have to capitalize the first letter.

**Review by LeeEnEn — APPROVED** [review]
Looks good to me.

File: src/main/java/seedu/address/model/candidate/Course.java (hunk 3 above)
Comment (domlimm) [reply src/main/java/seedu/address/model/candidate/Course.java:40]: @LeeEnEn Yes good point! I thought about this. Let me share with you my thoughts offline.

**Pull request merged**